go/tendermint/apps/staking/staking.go

```diff
@@ -112,48 +112,26 @@ func (app *stakingApplication) onEpochChange(ctx *abci.Context, epoch epochtime.
 			escrow = state.Account(e.EscrowID)
 		}
 
-		// Compute amount of debonded tokens.
-		tokens, err := staking.TokensForShares(&escrow.Escrow, &deb.Shares)
-		if err != nil {
-			app.logger.Error("failed to compute amount of tokens from shares",
+		var tokens staking.Quantity
+		if err := escrow.Escrow.Debonding.Withdraw(&tokens, &deb.Shares, &deb.Shares); err != nil {
```

Comment:
Is it ok if `Withdraw` is called with `srcShares` and `shareAmount` being the same pointer? Based on that function code it seems like there could be some unintended corruption.

Reply:
ah shoot no

Reply:
added Clone ):

Reply:
oops, had some other instances. fixed those too

Reply:
added tests in #2288